src/map/storage.c

```diff
@@ -200,69 +281,68 @@ static int storage_additem(struct map_session_data *sd, struct item *item_data,
 				it->amount += amount;
 
 				clif->storageitemadded(sd, it, i, amount);
-
-				sd->storage.save = true; // set a save flag.
+				stor->save = true; // set a save flag.
 
 				return 0;
 			}
 		}
 	}
 
 	// Check if storage exceeds limit.
-	if (sd->storage.aggregate >= MAX_STORAGE)
+	if (stor->aggregate >= stst->capacity)
 		return 1;
 
-	ARR_FIND(0, VECTOR_LENGTH(sd->storage.item), i, VECTOR_INDEX(sd->storage.item, i).nameid == 0);
+	ARR_FIND(0, VECTOR_LENGTH(stor->item), i, VECTOR_INDEX(stor->item, i).nameid == 0);
 
-	if (i == VECTOR_LENGTH(sd->storage.item)) {
-		VECTOR_ENSURE(sd->storage.item, 1, 1);
-		VECTOR_PUSH(sd->storage.item, *item_data);
-		it = &VECTOR_LAST(sd->storage.item);
+	if (i == VECTOR_LENGTH(stor->item)) {
+		VECTOR_ENSURE(stor->item, 1, 1);
```

Comment:
Memory manager reported a leak from this, I didn't check what is causing it (not sure if related to the missing VECTOR_CLEAR in unit.c

```
0001 : storage.c line 298 size 352 address 0x0x55976f3717e4
0002 : intif.c line 363 size 440 address 0x0x55977473f1fc
```